### PR TITLE
feat: revoke binding by resource type and resource id

### PR DESCRIPTION
**Pull request by mattia-fumo**

### CODE REVIEW

File: service/standalone_apis.go [hunk 1]
```diff
@@ -59,12 +59,11 @@ func revokeHandler(w http.ResponseWriter, r *http.Request) {
...
 
 	resourceType := mux.Vars(r)["resourceType"]
-	if resourceType != "" && len(reqBody.ResourceIDs) == 0 {
-		utils.FailResponseWithCode(w, http.StatusBadRequest, "empty resources list", utils.GENERIC_BUSINESS_ERROR_MESSAGE)
-		return
-	}
-	if len(reqBody.Subjects) == 0 && len(reqBody.Groups) == 0 {
-		utils.FailResponseWithCode(w, http.StatusBadRequest, "empty subjects and groups lists", utils.GENERIC_BUSINESS_ERROR_MESSAGE)
+	isResourceIDsQuery := resourceType != "" && len(reqBody.ResourceIDs) > 0
+	isSubjectsOrGroupsQuery := len(reqBody.Subjects) > 0 || len(reqBody.Groups) > 0
```
Comment (Danielecina) [inline service/standalone_apis.go:63]: I propose to use **has**ResourceId

File: service/standalone_apis_test.go [hunk 2]
```diff
@@ -59,10 +59,7 @@ func TestRevokeHandler(t *testing.T) {
 	})
 
 	t.Run("400 on missing resourceIds from body if resourceType request param is present", func(t *testing.T) {
-		reqBody := setupRevokeRequestBody(t, RevokeRequestBody{
-			Subjects: []string{"piero"},
-			Groups:   []string{"litfiba"},
-		})
+		reqBody := setupRevokeRequestBody(t, RevokeRequestBody{})
```
Comment (fredmaggiowski) [inline service/standalone_apis_test.go:62]: can you add a test to verify 400 is received if everything is an empty list? does it make sense? (it does to me lol)

File: service/standalone_apis_test.go [hunk 3]
```diff
@@ -177,6 +174,52 @@ func TestRevokeHandler(t *testing.T) {
...
+				Resource: &types.Resource{
+					ResourceType: "project",
+					ResourceID:   "mike",
+				},
+			},
+		}
+		gock.DisableNetworking()
+		newGockScope(t, "http://crud-service", http.MethodGet, "/bindings/").
+			AddMatcher(func(req *http.Request, greq *gock.Request) (bool, error) {
+				expected := `{"$and":[{"resource.resourceId":{"$in":["mike"]},"resource.resourceType":"myResource"},{"$or":[]}]}`
```
Comment (fredmaggiowski) [inline service/standalone_apis_test.go:191]: why the `{"$or":[]}` ? can we remove this query token?

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

File: service/standalone_apis.go (hunk 1 above)
Comment (mattia-fumo) [reply service/standalone_apis.go:63]: okay

File: service/standalone_apis_test.go (hunk 2 above)
Comment (mattia-fumo) [reply service/standalone_apis_test.go:62]: Shouldn't it be already covered by this test? By sending `RevokeRequestBody{}` each array is deserialized as a `nil` pointer that from a `len` point of view is the same as and empty list. Am I wrong?

Comment (fredmaggiowski) [reply service/standalone_apis_test.go:62]: yes, right!

File: service/standalone_apis_test.go (hunk 3 above)
Comment (mattia-fumo) [reply service/standalone_apis_test.go:191]: Sure. It is still there because i haven't touched the query generation logic

Comment (mattia-fumo) [reply service/standalone_apis_test.go]: Empty or removed

**Review by fredmaggiowski — APPROVED** [review]
